### PR TITLE
Nhat

### PR DESCRIPTION
**Pull request by Quang-Nhat-2601**

### CODE REVIEW

**Review by doonamm — COMMENTED** [review]
First review

File: app/Http/Controllers/ActorController.php [hunk 1]
```diff
@@ -3,8 +3,86 @@
...
+use App\Models\Actor;
 
 class ActorController extends Controller
 {
-    //
+    public function create(Request $request)
+    {
+        $validator = Validator::make($request->all(), [
+            'name' => 'required|string',
+            'popularity' => 'numeric',
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/ActorController.php:15]: ok

File: app/Http/Controllers/ActorController.php [hunk 2]
```diff
@@ -3,8 +3,86 @@
...
+        return response()->json([
+            'success' => true,
+            'message' => 'Create Actor Successfully',
+            'data' => $actor,
+        ]);
+    }
+
+    public function update(Request $request, Actor $actor)
+    {
+        $actor = Actor::findOrFail($actor);
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/ActorController.php:38]: ok

File: app/Http/Controllers/ActorController.php [hunk 3]
```diff
@@ -3,8 +3,86 @@
...
+        return response()->json([
+            'success' => true,
+            'message' => 'Delete Actor Successfully',
+        ]);
+    }
+
+    public function show(Request $request, Actor $actor)
+    {
+        return response()->json([
+            'data' => Actor::find($actor),
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/ActorController.php:74]: ok

File: app/Http/Controllers/CastController.php [hunk 4]
```diff
@@ -2,9 +2,85 @@
...
+
+        $cast = Cast::create($request->validated());
+
+        return response()->json([
+            'success' => true,
+            'message' => 'Create Cast Successfully',
+            'data' => $cast,
+        ]);
+    }
+
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/CastController.php:34]: ok

File: app/Http/Controllers/CastController.php [hunk 5]
```diff
@@ -2,9 +2,85 @@
...
+        return response()->json([
+            'success' => true,
+            'message' => 'Delete Cast Successfully',
+        ]);
+    }
+
+    public function show(Request $request, Cast $cast)
+    {
+        $cast = Cast::findOrFail($cast);
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/CastController.php:70]: ok

File: app/Http/Controllers/CommentController.php [hunk 6]
```diff
@@ -9,35 +9,95 @@
...
-            ],
+            'messsage' => 'Create Comment Successfully!',
+            'data' => $comment,
+        ]);
+    }
+
+    public function destroy(Request $request, Comment $comment)
+    {
+
+        $user = JWTAuth::toUser($request->bearerToken());
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/CommentController.php:47]: ok

File: app/Http/Controllers/CommentController.php [hunk 7]
```diff
@@ -9,35 +9,95 @@
...
+
+        $comment->delete();
+
+        return response()->json([
+            'success' => true,
+            'message' => 'Delete Comment Successfully',
+        ]);
+    }
+
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/CommentController.php:56]: ok

File: app/Http/Controllers/GenreController.php [hunk 8]
```diff
@@ -17,22 +17,41 @@ public function create(Request $request)
...
 
         return response()->json([
             'success' => true,
-            'message' => 'Create Gerne successful',
+            'message' => 'Create Genre Successful',
+            'data' => $genre,
         ]);
     }
 
-    public function update()
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/GenreController.php]: ok

File: app/Http/Controllers/CommentController.php [hunk 9]
```diff
@@ -9,35 +9,95 @@
...
+        $comment->update($request->all());
+
+        return response()->json([
+            'success' => true,
+            'message' => 'Update Comment Successfully',
+            'data' => $comment,
+        ]);
+    }
+
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/CommentController.php:85]: ok

File: app/Http/Controllers/CastController.php [hunk 10]
```diff
@@ -2,9 +2,85 @@
...
 
+use App\Models\Cast;
+use Illuminate\Support\Facades\Validator;
+
 use Illuminate\Http\Request;
 
 class CastController extends Controller
 {
-    //
+    public function create(Request $request)
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/CastController.php:12]: Cần nhận thêm param Actor không ông?

File: app/Http/Controllers/GenreController.php [hunk 11]
```diff
@@ -41,7 +60,25 @@ public function destroy(Request $request, Genre $genre)
 
         return response()->json([
             'success' => true,
-            'message' => 'Delete genre successfully',
+            'message' => 'Delete Genre Successfully',
+        ]);
+    }
+
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/GenreController.php:66]: ok

File: app/Http/Controllers/LikeController.php [hunk 12]
```diff
@@ -14,42 +14,69 @@ class LikeController extends Controller
 {
     public function create(Request $request)
     {
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/LikeController.php:16]: ok

File: app/Http/Controllers/LikeController.php [hunk 13]
```diff
@@ -14,42 +14,69 @@ class LikeController extends Controller
...
 
         return response()->json([
             'success' => true,
-            'message' => 'Create Like successfully',
+            'message' => 'Create Like Successfully',
+            'data' => $like,
         ]);
     }
 
-    public function update(Request $request)
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/LikeController.php]: ok

File: app/Http/Controllers/LikeController.php [hunk 14]
```diff
@@ -14,42 +14,69 @@ class LikeController extends Controller
...
         $like->delete();
 
         return response()->json([
             'success' => true,
-            'message' => 'Delete like successfully'
+            'message' => 'Delete Like Successfully'
         ]);
     }
 
-    public function show(Request $request, User $user, Comment $comment)
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/LikeController.php]: ok

File: app/Http/Controllers/LikeController.php [hunk 15]
```diff
@@ -14,42 +14,69 @@ class LikeController extends Controller
...
+                ->where('user_id', $request->user)
+                ->where('comment_id', $request->comment)
                 ->get(),
             'success' => true,
-            'message' => 'Delete profile successfully'
+            'message' => 'Show Like Successfully'
+        ]);
+    }
+
+    public function showAll(Request $request)
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/LikeController.php:74]: ok

File: app/Http/Controllers/MovieController.php [hunk 16]
```diff
@@ -4,34 +4,113 @@
...
+
+        if ($validator->fails()) {
+            return response()->json([
+                'success' => false,
+                'message' => 'Create Movie Failed',
+                'error' => $validator->errors()->toArray(),
+            ]);
+        }
+
+        $movie = Movie::query()->create([$request->validated()]);
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/MovieController.php:41]: ok

File: app/Http/Controllers/MovieController.php [hunk 17]
```diff
@@ -4,34 +4,113 @@
...
+        return response()->json([
+            'success' => true,
+            'message' => 'Update Movie Successfully',
+            'data' => $movie,
+        ]);
+    }
+
+    public function destroy(Request $request, Movie $movie)
+    {
+        $user = JWTAuth::toUser($request->bearerToken());
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/MovieController.php:89]: ok

File: app/Http/Controllers/MovieController.php [hunk 18]
```diff
@@ -4,34 +4,113 @@
...
+        return response()->json([
+            'success' => true,
+            'message' => 'Delete Movie Successfully'
+        ]);
+    }
+
+    public function show(Request $request, Movie $movie)
+    {
+        return response()->json([
+            'data' => Movie::find($movie),
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/MovieController.php:102]: ok

File: app/Http/Controllers/MovieGenreController.php [hunk 19]
```diff
@@ -5,12 +5,64 @@
 use Illuminate\Http\Request;
 use Tymon\JWTAuth\Facades\JWTAuth;
 use Illuminate\Support\Facades\Validator;
+use App\Models\Movie_Genre;
 
 
 class MovieGenreController extends Controller
 {
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/MovieGenreController.php:12]: ok

File: app/Http/Controllers/MovieGenreController.php [hunk 20]
```diff
@@ -5,12 +5,64 @@
...
+            'genre_id' => $request->input('genre_id'),
+        ]);
+
+        return response()->json([
+            'success' => true,
+            'message' => 'Create Movie Genre Successfully',
+            'data' => $movie_Genre,
+        ]);
+    }
+
```
[the remark this answers is not in the document]
Comment (Quang-Nhat-2601) [reply app/Http/Controllers/MovieGenreController.php:26]: ok

**Pull request closed without merging**